src/components/DataElement/Recent.jsx

```diff
@@ -10,14 +13,46 @@ import {
 import {Label3, Paragraph3, Paragraph4} from 'baseui/typography';
 import {StyledLink} from 'baseui/link';
 import {ProgressBar} from 'baseui/progress-bar';
-import {useStyletron} from 'baseui';
 
-export default function Recent({isLoading, data}) {
+function SearchIcon() {
+  const [css, theme] = useStyletron();
+  return (
+    <div
+      className={css({
+        display: 'flex',
+        alignItems: 'center',
+        paddingLeft: theme.sizing.scale500,
+      })}
+    >
+      <Search size="18px" />
+    </div>
+  );
+}
+
+const compare = (text = '', filterText = '') =>
+  text.toLowerCase().includes(filterText.toLowerCase());
+
+export default function Recent({ isLoading, data }) {
   const [, theme] = useStyletron();
+  const [filter, setFilter] = useState('');
+  const filteredData = data
+    ? data.filter(
+        ({ city, date }) => compare(city, filter) || compare(date, filter)
+      )
+    : [];
```

Comment:
I would probably add this part to `useEffect` hook with `filter` and `data` as dependants and then create `filteredData` as state value. I think that's the way it should be done, but I'm not sure.

Reply:
I've rewritten the code to useEffect. To be honest I'm not sure if it should be done this way, because filtering (already fetched) data doesn't sound like side-effect. In my opinion now the code is a little overcomplicated (although it works ;)). Official docs for React provides different examples of using useEffect, here is good summary of useEffect - https://typeofweb.com/react-hooks-useeffect-efekty-uboczne-w-komponencie/

You can decide what to do, I can revert a commit with useEffect as well.

Reply:
Hmm, I was thinking about the situations when the data passed to `Recent` component gets changed - which is possible due to live update. Will it work then?

Reply:
Correct me if I'm wrong, but I think the flow is following:
1. Data is being fetched by `DataElement` and passed to `Recent` component.
2. If data has changed, it means that `Recent`'s `data` prop has changed, so this component will perform rerender.
So in this scenario it should work.

Reply:
Yes, you're probably right. If you can revert this change that would be great. Sorry. :/

Reply:
That's all right! :) I have reverted the commit and added trimming filter value, e.g. user type " Wrocław " with spaces and filtering still works.